src/frontend/Komponenter/Oppgavebenk/OppgaveFiltrering.tsx

```diff
@@ -98,25 +116,17 @@ const OppgaveFiltrering: React.FC<IOppgaveFiltrering> = ({
     }, [oppgaveRequest.opprettetFom, oppgaveRequest.opprettetTom]);
 
     useEffect(() => {
-        const fraLocalStorage = hentFraLocalStorage(
+        const fraLocalStorage = hentFraLocalStorage<IOppgaveRequest>(
             oppgaveRequestKey(innloggetSaksbehandler.navIdent),
             {}
         );
-        if (fraLocalStorage) {
-            if (harSaksbehandlerStrengtFortroligRolle) {
-                settOppgaveRequest({
-                    ...fraLocalStorage,
-                    enhet: FortroligEnhet.VIKAFOSSEN,
-                });
-            } else {
-                settOppgaveRequest(fraLocalStorage);
-            }
-            hentOppgaver(fraLocalStorage);
-        } else {
-            if (harSaksbehandlerStrengtFortroligRolle) {
-                settOppgaveRequest({ enhet: FortroligEnhet.VIKAFOSSEN });
-            }
-        }
+
+        const oppgaveRequestMedEnhet = oppgaveRequestMedDefaultEnhet(
+            fraLocalStorage,
+            harSaksbehandlerStrengtFortroligRolle
+        );
+        settOppgaveRequest(oppgaveRequestMedEnhet);
+        hentOppgaver(oppgaveRequestMedEnhet);
```

Comment:
Noetere at if (fraLocalStorage) { fjernes då den alltid settes til true.
Så i stedet for å bruke hentOppgaver(fraLocalStorage); så hentes oppgaver med oppdatert enhet hentOppgaver(oppgaveRequestMedEnhet);